plugins/chat/app/models/chat_message.rb

```diff
@@ -91,20 +100,19 @@ def push_notification_excerpt
   end
 
   def to_markdown
-    markdown = []
+    upload_markdown =
+      self
+        .upload_references
+        .includes(:upload)
+        .order(:created_at)
+        .map { |upload_reference| UploadMarkdown.new(upload_reference.upload).to_markdown }
+        .reject(&:empty?)
 
-    if self.message.present?
-      msg = self.message
-
-      self.chat_uploads.any? ? markdown << msg + "\n" : markdown << msg
-    end
+    return self.message if upload_markdown.empty?
```

Comment:
Should this be a method on `upload_references` relation?

```
class UploadRefrence
  def self.to_markdown
    xxx
  end
end
```

?

Reply:
Actually yeah that will be cool, will do this :+1: